extensions/src/test/java/org/pentaho/platform/web/http/filters/ProxyTrustingFilterTest.java

```diff
@@ -246,5 +260,182 @@ public void doFilterForTrustedWithCustomLocaleHeader() throws Exception {
     assertEquals( new Locale( "pt", "PT" ), locale );
     assertEquals( locale.toString(), httpSession.getAttribute( IPentahoSession.ATTRIBUTE_LOCALE_OVERRIDE ) );
   }
+
+  @Test
+  public void testSetCsrfValidationRespectsIt() {
+
+    CsrfValidator csrfValidatorMock = mock( CsrfValidator.class );
+    ProxyTrustingFilter filter = new ProxyTrustingFilter(csrfValidatorMock);
+
+    // ---
+
+    assertSame( csrfValidatorMock, filter.getCsrfValidator() );
+  }
+
+  private void testWhenCsrfValidationFailsWithGivenExceptionThenRethrows( @NonNull Throwable validateError )
+          throws ServletException, IOException {
+
+    CsrfValidator csrfValidatorMock = mock( CsrfValidator.class );
+
+    MockFilterConfig cfg = new MockFilterConfig();
+    cfg.addInitParameter( "TrustedIpAddrs", "1.1.1.1," + TRUSTED_IP );
+
+    filter = new ProxyTrustingFilter(csrfValidatorMock);
+    filter.init( cfg );
+
+    request.setRemoteHost( TRUSTED_IP );
+    request.addParameter( filter.getParameterName(), "user" );
+
+    when( csrfValidatorMock.validateRequestOfMutationOperation( any( HttpServletRequest.class ),
+            eq( filter.getClass() ), anyString() ) )
+            .thenThrow( validateError );
+
+    // ---
+
+    filter.doFilter( request, new MockHttpServletResponse(), new MockFilterChain() );
+
+  }
+
+  @Test( expected = IOException.class )
+  public void testWhenCsrfValidationFailsWithIOExceptionThenRethrowsAsInternalAuthenticationServiceException() throws ServletException, IOException {
+
+    IOException error = mock( IOException.class );
+
+    try {
+
+      testWhenCsrfValidationFailsWithGivenExceptionThenRethrows(error);
+
+      fail( "Should have thrown exception" );
+    } catch ( InternalAuthenticationServiceException ex ) {
+        assertSame( error, ex.getCause() );
+      }
```

Comment:
The whole try/catch should not be here, including the `fail` and `assertSame` tests.
The `InternalAuthenticationServiceException` is not being thrown (anymore).
The `IOException` is being thrown directly from `doFilter` and from `testWhenCsrfValidationFailsWithGivenExceptionThenRethrows`.
The `@Test( expected = IOException.class )` is the only needed assertion.

Reply:
addressed review Comments